docs/content/testing/writing-a-new-test.md

```diff
@@ -0,0 +1,344 @@
+ is in the `Improbable.Gdk.Core.Utility` namespace.
+
+```cs
+using Improbable.Gdk.Core.Utility;
+using NUnit.Framework;
+
+namespace Improbable.Gdk.Core.EditmodeTests.Utility
+{
+    [TestFixture]
+    public class MyClassStaticTests
+    {
+    }
+
+    [TestFixture]
+    public class MyClassInstanceTests
+    {
+        private MyClass myClassInstance;
+
+        [SetUp]
+        public void SetUp()
+        {
+            myClassInstance = new MyClass();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            myClassInstance.Dispose();
+        }
+    }
+}
+```
+
+### Test methods
+
+Let's write a few test methods.
+
+The test should ensure that the `AddTwoNumbers` static method returns the sum of the
+ input numbers.
+
+The test name should match this expectation and contain relevant assertions.
+
+```cs
+    [TestFixture]
+    public class MyClassStaticTests
+    {
+        [Test]
+        public void AddTwoNumbers_should_return_the_sum_of_the_input_numbers()
+        {
+            Assert.AreEqual(4, MyClass.AddTwoNumbers(1, 3));
+            Assert.AreEqual(4, MyClass.AddTwoNumbers(3, 1));
+            Assert.AreEqual(7, MyClass.AddTwoNumbers(2, 5));
+        }
+    }
+```
+
+For the instance functionality, you can add tests similarly:
+
+```cs
+    [TestFixture]
+    public class MyClassInstanceTests
+    {
+        private MyClass myClassInstance;
+
+        [SetUp]
+        public void SetUp()
+        {
+            myClassInstance = new MyClass();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            myClassInstance.Dispose();
+        }
+
+        [Test]
+        public void InstanceValue_is_0_by_default()
+        {
+            Assert.AreEqual(0, myClassInstance.InstanceValue);
+        }
+
+        [Test]
+        public void DoInstanceOperation_increases_InstanceValue_by_1()
+        {
+            myClassInstance.DoInstanceOperation();
+
+            Assert.AreEqual(1, myClassInstance.InstanceValue);
+
+            myClassInstance.DoInstanceOperation();
+
+            Assert.AreEqual(2, myClassInstance.InstanceValue);
+        }
+
+        [Test]
+        public void DoSomethingElse_increses_InstanceValue_by_5()
+        {
+            myClassInstance.DoSomethingElse();
+
+            Assert.AreEqual(5, myClassInstance.InstanceValue);
+
+            myClassInstance.DoSomethingElse();
+
+            Assert.AreEqual(10, myClassInstance.InstanceValue);
+        }
+    }
+```
+
+### Internal variables and methods
+
+The tests above will not compile.
+
+This is because the test class is in another assembly than the implementation
+ class, and the tests are looking at internal fields.
+
+There is a solution. In the `MyClass.cs` file, you can make its internals
+ visible to the test assembly:
+
+```
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("Improbable.Gdk.Core.EditmodeTests")]
```

Comment:
I didn't know that! that's interesting (and makes life a lot easier) :D

Reply:
Credits to @samiwh who mentioned this to me :D